src/test/java/emissary/test/core/TestExtractionTest.java

```diff
@@ -51,6 +54,56 @@ void testCheckStringValueForCollectionFailure() throws JDOMException, IOExceptio
         assertThrows(AssertionError.class, () -> test.checkStringValue(meta, "7;0;0;0;2;1", "testCheckStringValueForCollection"));
     }
 
+    @Test
+    void testCheckStringValueBangIndex() throws IOException, JDOMException {
+        SAXBuilder builder = new SAXBuilder(XMLReaders.NONVALIDATING);
+        String resourceName = "/emissary/test/core/TestExtractionTest.xml";
+        InputStream inputStream = TestExtractionTest.class.getResourceAsStream(resourceName);
+        assertNotNull(inputStream, "Could not locate: " + resourceName);
+        Document answerDoc = builder.build(inputStream);
+        inputStream.close();
+        String matchMode = "!index";
+
+        WhyDoYouMakeMeDoThisExtractionTest test = new WhyDoYouMakeMeDoThisExtractionTest("nonsense");
+
+        List<Element> dataList = answerDoc.getRootElement().getChild("answers").getChildren("data");
+        Element data = getAttributeFromDataChild(dataList, matchMode);
+
+        test.checkStringValue(data, "time:20221229", "testCheckStringValue!IndexTrue");
+        assertThrows(AssertionError.class, () -> test.checkStringValue(data, "timestamp:20221229", "testCheckStringValue!IndexFalse"));
+
+    }
+
+    @Test
+    void testCheckStringValueBangContains() throws IOException, JDOMException {
+        SAXBuilder builder = new SAXBuilder(XMLReaders.NONVALIDATING);
+        String resourceName = "/emissary/test/core/TestExtractionTest.xml";
+        InputStream inputStream = TestExtractionTest.class.getResourceAsStream(resourceName);
+        assertNotNull(inputStream, "Could not locate: " + resourceName);
+        Document answerDoc = builder.build(inputStream);
+        inputStream.close();
+        String matchMode = "!contains";
+
+        WhyDoYouMakeMeDoThisExtractionTest test = new WhyDoYouMakeMeDoThisExtractionTest("nonsense");
+
+        List<Element> dataList = answerDoc.getRootElement().getChild("answers").getChildren("data");
+        Element data = getAttributeFromDataChild(dataList, matchMode);
+
+        test.checkStringValue(data, "time:20221229", "testCheckStringValue!ContainsTrue");
+        assertThrows(AssertionError.class, () -> test.checkStringValue(data, "timestamp:20221229", "testCheckStringValue!ContainsFalse"));
+    }
+
+    private Element getAttributeFromDataChild(List<Element> dataList, String matchMode) {
+        Element data = null;
+        // Having different matchModes in the same data necessitates having to go through each child and filter for the correct one
```

Comment:
To add to this comment, the other approach would be to make a different file for each matchMode we want to test, but that seemed cumbersome. If that's the preferred way to do it, then I'll change what I'm doing here.

Reply:
I recommend updating an existing answer file to exercise the that alias works should be sufficient. Maybe consider `sample.xml`.

Reply:
Removed `contains` tests, updated `HtmlEscapePlaceTest/sample.xml`.